tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -520,43 +560,66 @@ impl<'a, N> fmt::Display for FmtCtx<'a, N> {
     }
 }
 
-struct FullCtx<'a, N> {
-    ctx: &'a span::Context<'a, N>,
+pub struct FullCtx<'a, S, N>
+where
+    S: Subscriber + for<'lookup> LookupSpan<'lookup> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+{
+    ctx: &'a FmtContext<'a, S, N>,
     #[cfg(feature = "ansi")]
     ansi: bool,
 }
 
-impl<'a, N: 'a> FullCtx<'a, N> {
+impl<'a, S, N: 'a> FullCtx<'a, S, N>
+where
+    S: Subscriber + for<'lookup> LookupSpan<'lookup> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+{
     #[cfg(feature = "ansi")]
-    pub(crate) fn new(ctx: &'a span::Context<'a, N>, ansi: bool) -> Self {
+    pub(crate) fn new(ctx: &'a FmtContext<'a, S, N>, ansi: bool) -> Self {
         Self { ctx, ansi }
     }
 
     #[cfg(not(feature = "ansi"))]
-    pub(crate) fn new(ctx: &'a span::Context<'a, N>) -> Self {
+    pub(crate) fn new(ctx: &'a FmtContext<'a, S, N>) -> Self {
         Self { ctx }
     }
 }
 
 #[cfg(feature = "ansi")]
-impl<'a, N> fmt::Display for FullCtx<'a, N> {
+impl<'a, S, N> fmt::Display for FullCtx<'a, S, N>
+where
+    S: Subscriber + for<'lookup> LookupSpan<'lookup> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+{
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let mut seen = false;
         let style = if self.ansi {
             Style::new().bold()
         } else {
             Style::new()
         };
-        self.ctx.visit_spans(|_, span| {
-            write!(f, "{}", style.paint(span.name()))?;
-
+        self.ctx.visit_spans(|id| {
+            if seen {
+                f.pad(":")?;
+            }
             seen = true;
 
-            let fields = span.fields();
-            if !fields.is_empty() {
-                write!(f, "{}{}{}", style.paint("{"), fields, style.paint("}"))?;
+            let span = self.ctx.ctx.span(id);
+            if let Some(span) = span {
+                let ext = span.data.extensions_mut();
+                let data = ext.get::<String>().unwrap();
+                write!(f, "{} ", data);
```

Comment:
the old formatter put `{` `}` around the fields, which i personally like (esp. when we don't also have ansi formatting)

Reply:
Re-introduced via `write!(f, "{{{}}} ", data);`.

Reply:
(didn't push, but thank you for the reminder)